src/main/java/com/semmle/jira/addon/LgtmServlet.java

```diff
@@ -36,12 +37,14 @@
 
   @ComponentImport private final PluginSettingsFactory pluginSettingsFactory;
   @ComponentImport private final TransactionTemplate transactionTemplate;
+  private final PluginSettings settings;
 
   @Inject
   LgtmServlet(
       PluginSettingsFactory pluginSettingsFactory, TransactionTemplate transactionTemplate) {
     this.pluginSettingsFactory = pluginSettingsFactory;
     this.transactionTemplate = transactionTemplate;
+    this.settings = pluginSettingsFactory.createGlobalSettings();
```

Comment:
You probably need to mock this line of code, otherwise you most likely end-up with a null-valued `settings` field.

Reply:
There is a mock `PluginSettingsFactory` defined in `TestLgtmServletBase`, however the call to `createGlobalSettings` is not mocked, so mockito will make it return `null` by default.

Reply:
See for example https://github.com/Semmle/lgtm-jira-addon/blob/master/src/test/java/com/semmle/jira/addon/workflow/LgtmTransitionNotificationFunctionTest.java#L55 . That tests contains a simple implementation of the PluginSettings interface named `MockPluginSettings`. You might want to extract `MockPluginSettings` to its own file and use it too in the `TestCreateIssue` test.

Reply:
Ah, I was trying
```java
PluginSettings settings = mock(PluginSettings.class);
when(settingsFactory.createGlobalSettings()).thenReturn(settings);
```
And it wasn't working, kept getting `null`, as you said. I'll try `new MockPluginSettings()` instead!

Reply:
That should also work, but requires you to mock the method calls on the setting instance too.

Reply:
Yeah, I think there was something else I was still missing. But it became a moot point, as changed the approach to use managed config item, as per your suggestion... 👇